storey/aggregations.py

```diff
@@ -620,22 +619,26 @@ def get_features(self, timestamp, windows=None):
 
         return result
 
-    def calculate_features(self, timestamp):
+    def calculate_features(self, timestamp, windows=None):
         result = {}
 
         current_time_bucket_index = self.get_bucket_index_by_timestamp(timestamp)
         if current_time_bucket_index < 0:
             return result
 
-        all_windows = []
-        if self.explicit_windows:
-            some_window = self.explicit_windows
-            all_windows.extend(self.explicit_windows.windows)
-        if self.hidden_windows:
+        if windows:
+            all_windows = windows
             some_window = self.hidden_windows
-            for win in self.hidden_windows.windows:
-                if win not in all_windows:
-                    all_windows.append(win)
+        else:
+            all_windows = []
+            if self.explicit_windows:
+                some_window = self.explicit_windows
+                all_windows.extend(self.explicit_windows.windows)
+            if self.hidden_windows:
```

Comment:
Is it still possible to get here?
aren't you either don't have `windows` which means you use explicit windows or you have window (which are some of the hidden) and you use them?

Reply:
when coming from AggregationBuckets:__init__